site/components/src/main/java/uk/nhs/digital/common/components/apicatalogue/ApiCatalogueComponent.java

```diff
@@ -56,11 +62,33 @@ private Session sessionFrom(final HstRequest request) {
             .orElse(Collections.emptySet());
     }
 
-    private List<?> apiCatalogueLinksFrom(final HstRequest request) {
-        return ((ComponentList) request.getRequestContext().getContentBean()).getBlocks();
+    private List<Internallink> apiCatalogueLinksFrom(final HstRequest request, final Boolean isToShowAll) {
+
+        return isToShowAll ? retrieveAllLinks(request) : removeDeprecatedAndRetiredLinks(request);
+    }
+
+    private List<Internallink> retrieveAllLinks(HstRequest request) {
+
+        return ((ComponentList) request.getRequestContext().getContentBean()).getBlocks().stream()
+                .filter(Internallink.class::isInstance)
+                .map(Internallink.class::cast)
+                .collect(toList());
+    }
+
+    private List<Internallink> removeDeprecatedAndRetiredLinks(HstRequest request) {
+
+        List<?> blocks = ((ComponentList) request.getRequestContext().getContentBean()).getBlocks();
+
+        return blocks.stream()
+            .filter(Internallink.class::isInstance)
+            .map(Internallink.class::cast)
+            .filter(link -> !linkHasSelectedTaxonomyKeys(link, ImmutableSet.of("deprecated")))
+            .filter(link -> !linkHasSelectedTaxonomyKeys(link, ImmutableSet.of("retired")))
```

Comment:
Given that we have two keys to be excluded, how about passing them as a set and invoke the method just once? After all, it already supports a collection.